request_options.go

```diff
@@ -0,0 +1,95 @@
+package recurly
+
+import (
+	"context"
+	"net/http"
+)
+
+// RequestOptions contains additional request parameters for the API client
+type RequestOptions struct {
+	// IdempotencyKey used to prevent duplicate requests
+	IdempotencyKey string `json:"-"`
+	// Header contains additional request headers for unique requests
+	Header http.Header `json:"-"`
+	// Context passed to the HTTP request for cancelling requests
+	Context context.Context `json:"-"`
+}
+
+func (o *RequestOptions) clone() *RequestOptions {
+	clone := &RequestOptions{}
+	clone.Header = o.Header.Clone()
+	clone.IdempotencyKey = o.IdempotencyKey
+	clone.Context = o.Context
```

Comment:
since context is an interface, copying it here will copy a pointer to the same struct. that means if the requestOptions that is cloned cancels the context, the context will also be cancelled on the parent.